installers/binstall/src/system/unix.rs

```diff
@@ -0,0 +1,212 @@
+use std::env;
+use std::fmt::Debug;
+use std::fs;
+use std::io::{self, Write};
+use std::path::PathBuf;
+
+use crate::{Installer, InstallerError};
+
+pub fn add_binary_to_path(installer: &Installer) -> Result<(), InstallerError> {
+    let mut written = vec![];
+
+    for shell in get_available_shells() {
+        let source_cmd = shell.source_string(installer)?;
+
+        for rc in shell.update_rcs() {
+            if !rc.is_file() || !fs::read_to_string(&rc)?.contains(&source_cmd) {
+                tracing::debug!("updating {}", &rc.display());
+                let mut dest_file = fs::OpenOptions::new()
+                    .write(true)
+                    .append(true)
+                    .create(true)
+                    .open(&rc)?;
+                writeln!(&mut dest_file, "{}", &source_cmd)?;
+                dest_file.sync_data()?;
+
+                let script = shell.env_script();
+
+                if !written.contains(&script) {
+                    script.write(installer)?;
+                    written.push(script);
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn get_available_shells() -> impl Iterator<Item = Shell> {
+    let shells: Vec<Shell> = if cfg!(test) {
+        vec![]
+    } else {
+        vec![Box::new(Posix), Box::new(Bash), Box::new(Zsh)]
```

Comment:
does fish get handled by posix?

Reply:
fish is not currently supported, those users will have to add to their Path themselves (rustup [does not do this](https://github.com/rust-lang/rustup/issues/478) for fish users right now since i think it's a little strange.)

i think fish users are pretty used to this, though i'm not opposed to adding support (dont think we need it for v0.1 though)